build.sh

```diff
@@ -189,20 +291,25 @@ function parseArgs() {
     done
 }
 
+function checkJava() {
+    if ! command -v   java &> /dev/null ; then
+        err_log "It seems you do not have java installed. Please ensure you have it installed and executable as \"java\"."
+        exit 1
+    fi
+}
+
 function checkPreconditions() {
     if ! command -v   mvn &> /dev/null ; then
         err_log "It seems you do not have maven installed. Please ensure you have it installed and executable as \"mvn\"."
         exit 1
     fi
 
-    if ! command -v   java &> /dev/null ; then
-        err_log "It seems you do not have java installed. Please ensure you have it installed and executable as \"java\"."
-        exit 1
-    fi
+    checkJava
 
     BASEDIR=$(mvn help:evaluate -Dexpression=project.build.directory --non-recursive -q -DforceStdout)
+    JMC_DIR=$(pwd)
     mkdir -p "${BASEDIR}" # just in case clean was called before
 }
 
 checkPreconditions
```

Comment:
Thinking about it now, should checkPreconditions happen here, or would it be better to have it before valid arguments in `parseArgs`?

My reasoning being if I just wanted to just see the help menu, the script has to first check for java & maven, and then prints the menu. It's not a particularly long delay, but there is a delay while it verifies. I also noticed this because I had JAVA_HOME unset so the script failed without any output even though I just wanted to read the options.

Reply:
I've changed the checks order, JAVA_HOME is 1st now. Nice catch !